### PR TITLE
Fix onSpawn return false and missing creature id errors

### PR DESCRIPTION
**Pull request by Codinablack**
Co-Authored-By: Forgee <3932953+forgee@users.noreply.github.com>
 Special thanks to @MillhioreBT for her advice on where to put monster->decrementReferenceCounter();

### Pull Request Prelude

- [x] I have followed [proper The Forgotten Server code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR may be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed

[#4086]
[#4077]

This fixes the monster not being assigned a creature id during "onSpawn" event, it safely removes the monster if returned false, and fixes the problem of returning false with onSpawn breaking the internal spawn check system. 

Spent much time finding #4086 when trying to solve #4077, but it was a problem unknown and this pr has been tested pretty well, no errors found. 

[cont]: https://github.com/otland/forgottenserver/wiki/Contributing
[code]: https://github.com/otland/forgottenserver/wiki/TFS-Coding-Style-Guide

### CODE REVIEW

**Review by ramon-bernardo — APPROVED** [review]
LGTM

**Comment by amatria** [comment]
Just out of curiosity, why is the `spawnMonster` method creating a `std::unique_pointer` to immediately release the ownership of the stored object? Why bother creating an auto managed pointer if it is not going to be used at all? Am I missing something?

**Comment by nekiro** [comment]
> Just out of curiosity, why is the `spawnMonster` method creating a `std::unique_pointer` to immediately release the ownership of the stored object? Why bother creating an auto managed pointer if it is not going to be used at all? Am I missing something?

To avoid memory leaks, when placeCreature fails

File: src/spawn.cpp [hunk 1]
```diff
@@ -336,6 +333,12 @@ bool Spawn::spawnMonster(uint32_t spawnId, MonsterType* mType, const Position& p
 	monster->setSpawn(this);
 	monster->setMasterPos(pos);
 	monster->incrementReferenceCounter();
+	if (!g_events->eventMonsterOnSpawn(monster, pos, startup, false)) {
```
Comment (nekiro) [inline src/spawn.cpp:336]: not a fan of this, placing a monster to delete it, sounds like a waste

Comment (Codinablack) [reply src/spawn.cpp:336]: Then return false won't be fixed. 🤷‍♂️

**Pull request closed without merging**